src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -163,6 +164,40 @@ public UserChallenge initUserChallenge(Challenge challenge, Invitation invitatio
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public List<ChallengeDTO> getChallengesWithPendingStatusForInvitation(Long userId) throws NotFoundException {
+
+        List<UserChallenge> userChallengeList = userChallengeRepository.findAllByUserId(userId);
```

Comment:
Ma gandesc ca o metoda din repo care sa caute in baza de date dupa status este mult mai utila. Asa cum ai implementat o sa incarci in memorie o lista de UserChallenge chiar daca nu ai nevoie de ele. 
Poti defini o metoda findByUserIdAndStatus care sa-ti aduca informatiile